src/server/bitops_family.cc

```diff
@@ -57,14 +65,88 @@ constexpr int32_t GetByteIndex(uint32_t offset) noexcept {
   return offset / OFFSET_FACTOR;
 }
 
-uint8_t GetByteValue(const std::string& str, uint32_t offset) {
+uint8_t GetByteValue(std::string_view str, uint32_t offset) {
   return static_cast<uint8_t>(str[GetByteIndex(offset)]);
 }
 
 constexpr bool CheckBitStatus(uint8_t byte, uint32_t offset) {
   return byte & (0x1 << offset);
 }
 
+constexpr std::uint8_t CountBitsRange(std::uint8_t byte, std::uint8_t from, uint8_t to) {
+  int count = 0;
+  for (int i = from; i < to; i++) {
+    count += CheckBitStatus(byte, GetNormalizedBitIndex(i));
+  }
+  return count;
+}
+
+constexpr std::uint8_t BitOnCount(std::uint8_t byte) {
+  return NIBBLE_LOOKUP[byte & 0x0F] + NIBBLE_LOOKUP[byte >> 4];
+}
+
+std::uint32_t CountSetAllBytes(std::string_view at, int32_t start, int32_t end) {
```

Comment:
lets use size_t when counting bits - even if we have 512M length strings, there result will be beyond uint32_t

Reply:
done